File: htmltmpl-php/test/compiled.php
<?

$TEST = "compiled";
if (in_array('debug', $argv)) {
    $HTMLTMPL_DEBUG="debug.log";
}
require('../htmltmpl.php');
$man = new TemplateManager();
$template =& $man->prepare("$TEST.tmpl");
$tproc = new TemplateProcessor();

#######################################################

function fill(&$tproc) {
    $tproc->set("title", "Template world.");
    $tproc->set("greeting", "Hello !");
    $tproc->set("Boys", array(
        array( "name" => "Tomas",  "age" => 19 ),
        array( "name" => "Pavel",  "age" => 34 ),
        array( "name" => "Janek",  "age" => 67 ),
        array( "name" => "Martin", "age" => 43 ),
        array( "name" => "Viktor", "age" => 78 ),
        array( "name" => "Marian", "age" => 90 ),
        array( "name" => "Prokop", "age" => 23 ),
        array( "name" => "Honzik", "age" => 46 ),
        array( "name" => "Brudra", "age" => 64 ),
        array( "name" => "Marek",  "age" => 54 ),
        array( "name" => "Peter",  "age" => 42 ),
        array( "name" => "Beda",   "age" => 87 )
    ));
}

#######################################################

fill($tproc);
$tproc->process($template);
$tproc->reset();

fill($tproc);
$tproc->process($template);
$tproc->reset();

fill($tproc);
$output = $tproc->process($template);

if (in_array('out', $argv)) {
    echo $output;
    exit;
}

$file_res = fopen("$TEST.res", 'r');
$res = fread($file_res, filesize("$TEST.res"));
fclose($file_res);

echo "$TEST ... ";

if ($output == $res && is_file("$TEST.tmplc")) {
    echo "OK";
    unlink("$TEST.tmplc");
    unlink("$TEST.tmplcc");
}
else {
    echo "FAILED";
}
?>

File: htmltmpl/test/compiled.py
#!/usr/bin/env python

TEST = "compiled"

import sys
import os
sys.path.insert(0, "..")

from htmltmpl import Template

tmpl = Template(template = TEST + ".tmpl",
                template_path = ["."],
                compile = 1,
                cache = 0,
                debug = "debug" in sys.argv)

#######################################################

def fill(tmpl):
    tmpl["title"] = "Template world."
    tmpl["greeting"] = "Hello !"
    tmpl["Boys"] = [
        { "name" : "Tomas",  "age" : 19 },
        { "name" : "Pavel",  "age" : 34 },
        { "name" : "Janek",  "age" : 67 },
        { "name" : "Martin", "age" : 43 },
        { "name" : "Viktor", "age" : 78 },
        { "name" : "Marian", "age" : 90 },
        { "name" : "Prokop", "age" : 23 },
        { "name" : "Honzik", "age" : 46 },
        { "name" : "Brudra", "age" : 64 },
        { "name" : "Marek",  "age" : 54 },
        { "name" : "Peter",  "age" : 42 },
        { "name" : "Beda",   "age" : 87 }
    ]

#######################################################

fill(tmpl)
tmpl.output()
tmpl.reset()

fill(tmpl)
tmpl.output()
tmpl.reset()

fill(tmpl)
output = tmpl.output()

if "out" in sys.argv:
    sys.stdout.write(output)
    sys.exit(0)

res = open("%s.res" % TEST).read()

print TEST, "...",

if output == res and os.access("%s.tmplc" % TEST, os.R_OK):
    print "OK"
    os.remove("%s.tmplc" % TEST)
else:
    print "FAILED"
    open("%s.fail" % TEST, "w").write(output)
